include/swift/Reflection/ReflectionContext.h

```diff
@@ -378,30 +378,37 @@ class ReflectionContext
     return readPECOFFSections(ImageStart);
   }
 
-  template <typename T> bool readELFSections(RemoteAddress ImageStart) {
+  template <typename T>
+  bool readELFSections(RemoteAddress ImageStart, const void *FileBuffer,
+                       uint64_t FileSize) {
     auto Buf =
         this->getReader().readBytes(ImageStart, sizeof(typename T::Header));
 
     auto Hdr = reinterpret_cast<const typename T::Header *>(Buf.get());
     assert(Hdr->getFileClass() == T::ELFClass && "invalid ELF file class");
 
-    // From the header, grab informations about the section header table.
-    auto SectionHdrAddress = ImageStart.getAddressData() + Hdr->e_shoff;
+    auto FileBufferStartAddress = (uint64_t) FileBuffer;
+    // From the header, grab information about the section header table.
+    auto SectionHdrAddress = FileBufferStartAddress + Hdr->e_shoff;
     auto SectionHdrNumEntries = Hdr->e_shnum;
     auto SectionEntrySize = Hdr->e_shentsize;
 
+    if (FileBufferStartAddress + FileSize <
+        SectionHdrAddress + (SectionHdrNumEntries - 1) * SectionEntrySize +
```

Comment:
What happens if SectionHdrNumEntries is 0? Can we add some minimum amount of error handling for this? Early exit is fine.

Reply:
LLDB often ends up having to deal with some form of corrupted input and you don't want to crash when that happens.